src/librustc_mir/const_eval.rs

```diff
@@ -468,7 +468,7 @@ impl<'a, 'mir, 'tcx> interpret::Machine<'a, 'mir, 'tcx>
     }
 }
 
-/// Projects to a field of a (variant of a) const.
+/// Extracts a field of a (variant of a) const.
 // this function uses `unwrap` copiously, because an already validated constant must have valid
 // fields and can thus never fail outside of compiler bugs
 pub fn const_field<'a, 'tcx>(
```

Comment:
Our other queries here are called `*_provider`. Might make sense to be consistent about that?

Reply:
Using the name of the query is proper, IMO - however, it should definitely not be public!

Reply:
It is public because it is called directly elsewhere, as per @oli-obk... can’t find the comment atm.

Reply:
@eddyb ^^^